server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSExtractOptionsUtility.java

```diff
@@ -0,0 +1,255 @@
+          put("PATH", ErrorSeverity.ERROR);
+          put("PATHLENGTH", ErrorSeverity.ERROR);
+          put("HTTPMETHOD", ErrorSeverity.ERROR);
+          put("METHODLENGTH", ErrorSeverity.ERROR);
+          put("QUERYSTRING", ErrorSeverity.ERROR);
+          put("QUERYSTRLEN", ErrorSeverity.ERROR);
+          put("URIMAP", ErrorSeverity.ERROR);
+          put("SESSTOKEN", ErrorSeverity.ERROR);
+          put("REALM", ErrorSeverity.ERROR);
+          put("REALMLEN", ErrorSeverity.ERROR);
+        }
+      };
+
+  public CICSExtractOptionsUtility(DialectProcessingContext context, List<SyntaxError> errors) {
+    super(context, errors, DUPLICATE_CHECK_OPTIONS);
+  }
+
+  /**
+   * Entrypoint to check CICS Extract rule options
+   *
+   * @param ctx ParserRuleContext subclass containing options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    if (ctx.getClass() == CICSParser.Cics_extract_attachContext.class)
+      checkAttach((CICSParser.Cics_extract_attachContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_attributesContext.class)
+      checkAttributes((CICSParser.Cics_extract_attributesContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_certificateContext.class)
+      checkCertificate((CICSParser.Cics_extract_certificateContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_logonmessageContext.class)
+      checkLogonMsg((CICSParser.Cics_extract_logonmessageContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_processContext.class)
+      checkProcess((CICSParser.Cics_extract_processContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_tcpipContext.class)
+      checkTcpIp((CICSParser.Cics_extract_tcpipContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_tctContext.class)
+      checkTCT((CICSParser.Cics_extract_tctContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_web_clientContext.class)
+      checkWebClient((CICSParser.Cics_extract_web_clientContext) ctx);
+    else if (ctx.getClass() == CICSParser.Cics_extract_web_serverContext.class)
+      checkWebServer((CICSParser.Cics_extract_web_serverContext) ctx);
+    checkDuplicates(ctx);
+  }
+
+  private void checkAttach(CICSParser.Cics_extract_attachContext ctx) {
+    checkHasMandatoryOptions(ctx.ATTACH(), ctx, "ATTACH");
+    if (!ctx.ATTACHID().isEmpty()) {
+      checkHasIllegalOptions(ctx.CONVID(), "CONVID with ATTACHID");
+      checkHasIllegalOptions(ctx.SESSION(), "SESSION with ATTACHID");
+    } else if (!ctx.CONVID().isEmpty()) {
+      checkHasIllegalOptions(ctx.ATTACHID(), "ATTACHID with CONVID");
+      checkHasIllegalOptions(ctx.SESSION(), "SESSION with CONVID");
+    } else if (!ctx.SESSION().isEmpty()) {
+      checkHasIllegalOptions(ctx.ATTACHID(), "ATTACHID with SESSION");
+      checkHasIllegalOptions(ctx.CONVID(), "CONVID with ATTACHID");
+    }
```

Comment:
This might warrant creating a new test function that will check mutually exclusive options.

`checkMututallyExclusiveOptions("A, B, C", ctx.A(), ctx.B(), ctx.C())` -> diagnostic could be "Options {...} are mutually exclusive"